conda/Dockerfile

```diff
@@ -4,5 +4,7 @@ ARG ROOT_VERSION=6.20.00
 ARG PYTHON_VERSION=3.7
 
 RUN conda config --add channels conda-forge && \
-    conda install --quiet python=$PYTHON_VERSION root=$ROOT_VERSION && \
-    conda clean --all
+    conda config --set allow_softlinks false && \
```

Comment:
Pardon my lack of conda knowledge .. why does "not using soft links" reduce the image size?  It sounds like softlink would be replaced by copies and thus increase the size.  What am I missing?

Reply:
(Oversimplifying a little, but it explains the rough idea)

In conda there are directories containing environments `envs/$ENVIRONMENT_NAME` and a packages cache `pkgs/$PACKAGE_NAME_AND_HASH`. By default conda will try to hardlink or symlink files from the cache into the packages directory however it is only able to do this if there isn't a need to perform binary relocation tricks. It also seems to be quite conservative to avoid errors.

This means that there can be a lot of duplication files between `pkgs/` and `envs/`. If I'm making a specialised image I normally remove this bloat by doing a multi-stage docker build to copy the environment to minimal container.

However, in this case I want `conda install` to still be functional inside the container so `conda clean --force-pkgs-dirs` is ran at the end to effectively run `rm -rf pkgs/*`. This isn't done by default as it's a potentially unsafe operation if there is an untracked environment using links to these files. As setting `allow_softlinks=false` and `always_copy=true` prevents this, `pkgs/` becomes nothing more than a cache so we can get rid of it.

(I don't know if both options are actually needed, but I don't see any harm in being conservative.)

Reply:
Thanks.  That makes perfect sense.